### PR TITLE
proto: split proto-files

### LINKED ISSUE #2: Split `aeon_router.proto`
We decided to split the file into `aeon_sql.proto` and `aeon_crud.proto`. The `aeon_crud.proto` will be `experimental`.

### PR DESCRIPTION
**Pull request by ImeevMA**
This splits the proto file into three files: aeon_sql.proto, aeon_crud.proto, and aeon_common.proto.

This patch also adds a warning that CRUD operations are experimental.

Closes #2

### CODE REVIEW

File: aeon_sql.proto [hunk 1]
```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+import "aeon_common.proto";
+
+package aeon;
+
+option go_package = "./pb";
+
+// SQL API to Aeon - a distributed database based on Tarantool.
+service AeonSQLService {
```
Comment (locker) [inline aeon_sql.proto]: Let's remove the `Aeon` prefix: `SQLService`, `CRUDService`. The package name should serve as a namespace.

File: aeon_sql.proto [hunk 2]
```diff
@@ -0,0 +1,44 @@
+syntax = "proto3";
+
+import "aeon_common.proto";
+
+package aeon;
+
+option go_package = "./pb";
```
Comment (locker) [inline aeon_sql.proto]: Shouldn't we use a proper name here? Take a look at [YDB](https://github.com/ydb-platform/ydb-api-protos/blob/1440f00ea8c997bfe378c9e8e8a29c8916b8b4ba/ydb_monitoring_v1.proto#L5), for example.

File: aeon_common.proto [hunk 3]
```diff
@@ -0,0 +1,185 @@
+syntax = "proto3";
+
+package aeon;
+
+option go_package = "./pb";
+
+// Common API to Aeon - a distributed database based on Tarantool.
+service AeonCommonService {
```
Comment (locker) [inline aeon_common.proto]: I think we should put the `Ping` request into a separate service, for example, `DiagService` or something like it (I haven't really though about a name). Maybe, split the common proto into more parts (schema, value, tuple, error)?

Comment (unera) [reply aeon_common.proto]: i think it is worth to extract one other .proto: ddl.proto

Now we can move `create_space`/`drop_space` to the file.
Then we'll add `list_spaces` etc here.

File: aeon_sql.proto (hunk 1 above)
Comment (ImeevMA) [reply aeon_sql.proto]: Fixed.

File: aeon_sql.proto (hunk 2 above)
Comment (ImeevMA) [reply aeon_sql.proto]: Done.

File: aeon_common.proto (hunk 3 above)
Comment (ImeevMA) [reply aeon_common.proto]: Thanks, fixed. Now we have following files:
```
aeon_crud.proto
aeon_ddl.proto
aeon_diag.proto
aeon_error.proto
aeon_schema.proto
aeon_sql.proto
aeon_value.proto
```

File: aeon_crud.proto [hunk 4]
```diff
@@ -1,32 +1,17 @@
...
-  // Diagnostic requests.
-  //
-
-  // Pings the router.
-  rpc Ping(PingRequest) returns (PingResponse) {}
-
-  //
-  // DDL requests.
-  //
+option go_package = "github.com/tarantool/aeon/aeon/grpc/server/pb";
```
Comment (locker) [inline aeon_crud.proto]: Decided to remove this for now and instead provide a helper script that would generate pb.go files taking the package name as an argument.

File: generate-go.sh [hunk 5]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+if [[ -z "$1" ]]
+then
+  out_path="."
+else
+  out_path=$1
+fi
```
Comment (dmyger) [inline generate-go.sh]:
```suggestion
out_path=${1:-.}
```

File: generate-go.sh
Comment (dmyger) [inline generate-go.sh:1]: Not executable: `chmod +x generate-go.sh`

File: generate-go.sh [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+
+if [[ $out_path != .* ]] && [[ $out_path != /* ]]
+then
+  echo "Path must start with '.' or with '/'"
```
Comment (dmyger) [inline generate-go.sh]: Why it's required?

File: generate-go.sh [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+  exit 1
+fi
+
+if [[ -z "$2" ]]
+then
+  proto_path="."
+else
+  proto_path=$2
+fi
```
Comment (dmyger) [inline generate-go.sh]:
```suggestion
proto_path=${2:-.}
```

File: generate-go.sh [hunk 8]
```diff
@@ -0,0 +1,43 @@
+#!/usr/bin/env bash
+
+if [[ -z "$1" ]]
+then
+  out_path="."
```
Comment (locker) [inline generate-go.sh]: 1. Let's use `getopt` for parsing command line options.
2. Let's add two options: `--package <package>` and `--out <path>`. The first one specifies the Go package for generated files; it should be mandatory. The second option specifies the path to the directory where to store generated files; it should be optional.
3. No need to specify the path to the proto files. Let's deduce it automatically assuming that the files are stored in the same directory as `generate-go.sh` with `readlink` and `dirname`.
4. If the user calls the function without any arguments or passes an unknown option, let's print a help message, like this:
   ```
   Usage: $0 --package NAME [--out PATH]
   Generate Go bindings for Aeon proto files.
   
   Options:
     --package NAME: Go package name.
     --out PATH: Path to the output directory.
   ```

File: generate-go.sh (hunk 6 above)
Comment (locker) [inline generate-go.sh]: Let's write errors to stderr (`>2`).

File: generate-go.sh [hunk 9]
```diff
@@ -0,0 +1,43 @@
...
+
+if [[ -z "$2" ]]
+then
+  proto_path="."
+else
+  proto_path=$2
+fi
+
+protoc \
+--proto_path=$proto_path \
```
Comment (locker) [inline generate-go.sh]: Nit: Please add an indentation here.

Also, I'd try to generate the arguments from the file list.

File: generate-go.sh (hunk 5 above)
Comment (ImeevMA) [reply generate-go.sh]: Removed this part.

File: generate-go.sh
Comment (ImeevMA) [reply generate-go.sh:1]: Fixed.

File: generate-go.sh (hunk 6 above)
Comment (ImeevMA) [reply generate-go.sh]: Removed. This is requirement of `protoc` if `go_package` is not a URI.

File: generate-go.sh (hunk 7 above)
Comment (ImeevMA) [reply generate-go.sh]: Dropped.

File: generate-go.sh (hunk 8 above)
Comment (ImeevMA) [reply generate-go.sh]: Fixed.

File: generate-go.sh (hunk 6 above)
Comment (ImeevMA) [reply generate-go.sh]: Fixed.

File: generate-go.sh (hunk 9 above)
Comment (ImeevMA) [reply generate-go.sh]: Fixed.

**Review by unera — APPROVED** [review]

**Review by dmyger — APPROVED** [review]

File: generate-go.sh [hunk 10]
```diff
@@ -0,0 +1,58 @@
...
+done
+
+if [[ -z $package ]]
+then
+  usage
+fi
+
+if [[ -z $out ]]
+then
+  out=.
```
Comment (locker) [inline generate-go.sh:44]: Let's create the out directory with `mkdir -p` if it doesn't exist.

**Review by locker — APPROVED** [review]

File: generate-go.sh (hunk 10 above)
Comment (ImeevMA) [reply generate-go.sh:44]: Fixed.

**Pull request merged**